Cargo.toml

```diff
@@ -18,14 +18,14 @@ cfg-if = "1"
 compiler_builtins = { version = "0.1", optional = true }
 core = { version = "1.0", optional = true, package = "rustc-std-workspace-core" }
 
-[target.'cfg(unix)'.dependencies]
+[target.'cfg(all(unix, not(target_os = "linux")))'.dependencies]
 libc = { version = "0.2.154", default-features = false }
 
 [target.'cfg(target_os = "wasi")'.dependencies]
 wasi = { version = "0.11", default-features = false }
 
-[target.'cfg(any(target_os = "linux", target_os = "android"))'.dependencies]
-rustix = { version = "0.38.0", features = ["event", "fs", "rand", "thread"], default-features = false, optional = true }
+[target.'cfg(target_os = "linux")'.dependencies]
+rustix = { version = "0.38.0", features = ["event", "fs", "rand", "thread"], default-features = false }
```

Comment:
We should only need `event`, `fs`, and `thread` if we're going to use the file-based fallback, but unfortunately that's very hard to specify in Cargo.toml, as you can see by the actual cfg logic in lib.rs.

I guess there must be a reason why Rustix makes those things optional features. (Why?) Would people who don't want those features activated care that `getrandom` would be activating them?

Reply:
I don't think so, the features are mostly to reduce code size.